provider/cmd/pulumi-resource-aws/generate.go

```diff
@@ -65,10 +66,18 @@ func compressAndVersionSchemaFile(opts compressAndVersionSchemaFileOptions) erro
 	if err != nil {
 		log.Fatalf("failed to open file: %s", err)
 	}
-	defer file.Close()
+	defer func() {
+		err := file.Close()
+		contract.AssertNoErrorf(err, "Failed closing file")
+	}()
 	var w io.Writer = file
 	if opts.gzip {
-		w = gzip.NewWriter(file)
+		gzipWriter := gzip.NewWriter(file)
+		w = gzipWriter
+		defer func() {
+			err := gzipWriter.Close()
```

Comment:
So the issue was that the writer wasn't flushed before?

Reply:
Yes and it made garbled data.

Reply:
Got it!